api_types.ts

```diff
@@ -995,14 +996,18 @@ export interface ExecutionContext {
   readonly authenticationName?: string;
 
   /**
-   * If this invocation is a part of a crawling execution, like in Coda Brain, then this ID will be provided
+   * If this invocation is a part of a crawl sync, then this ID will be provided
    * to all invocations. That includes invocations of sync `execute` and `executeGetPermissions`, as well as
    * dynamic table features like `listDynamicUrls`, `getSchema`, and `getName`.
    *
+   * This may be a full sync ID or an incremental sync ID.
+   *
+   * TODO(patrick): May want to support this in doc syncs too.
+   *
    * TODO(patrick): Unhide this
    * @hidden
    */
-  readonly executionId?: string;
+  readonly syncId?: string;
```

Comment:
Is this called simply "sync id" instead of "crawlSyncId" or similar because of "TODO(patrick): May want to support this in doc syncs too."?

Reply:
Option1: `syncExecutionId`
Option2: `executionId` and move into `sync`, verifying that `sync` is passed into metadata formulas

Decision: Option2 is preferred

official term is "full/incremental sync execution"